pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -181,6 +198,55 @@ protected void removeSegment(IndexSegment segment, MutableRoaringBitmap validDoc
     }
   }
 
+  /**
+   * When TTL is enabled for upsert, this function is used to remove expired keys from the primary key indexes.
+   *
+   * When committing consuming segment, we replace the consuming segment with an immutable segments.
+   * After replaceSegment, we iterate over recordInfoIterator to find validDocIds that are expired (out-of-TTL).
+   * Primarykey expired when the comparison time value of the record is less or equal to (segmentEndTime - TTL).
+   *
+   * @param expiredTimestamp segmentEndTime - TTLTime (converted ttl time values in millis time unit)
+   * @return void
+   */
+  @Override
+  public void doRemoveExpiredPrimaryKeys(Comparable expiredTimestamp) {
```

Comment:
Currently this works only when the comparison column is timestamp (time column in milliseconds). We should add some validation to the table config

Reply:
Yes, plus now we have support for multiple comparison columns as well. We can enforce a requirement for now that if TTL is enabled only one comparison column can be there and it needs to be timestamp.

Reply:
+1 to the above. can you please call this out in the javadoc and in the oss pinot docs (when you get to it)?

Reply:
Sure, will do